### PR TITLE
Feature/#193 create beneficiary

### PR DESCRIPTION
**Pull request by mayapeneva**
Added CreateBeneficiary endpoint.

### CODE REVIEW

**Comment by kachar** [comment]
@mayapeneva Is this one ready for review / merge ?

**Comment by xenry** [comment]
looking very good. Lets just add an organizerId to the beneficiary so that we know who created it.

File: backend/protos/campaign.proto [hunk 1]
```diff
@@ -80,6 +83,22 @@ message CampaignTypesResponse {
 	repeated CampaignTypeDetails campaign_types = 1;
 }
 
+message CreateBeneficiaryRequest {
+
+	string name = 1;
+	common.nomenclatures.BeneficiaryType type = 2;
+	common.nomenclatures.ISO2CountryCode country_iso_code = 3;
+  string city = 4;
```
Comment (xenry) [inline backend/protos/campaign.proto:91]: formatting here can be fixed, seems that some of the fields are indented with tabs

File: db/migrations/campaign/V4.0.0.4__table_campaigns.sql [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+CREATE TABLE IF NOT EXISTS campaign.beneficiaries (
+    id uuid NOT NULL DEFAULT (gen_random_uuid()),
+    name character varying(100) NOT NULL,
+    type integer NOT NULL,
+    iso2country_code integer NOT NULL,
+    city character varying(50) NOT NULL,
+    email character varying(100) NULL,
+    phone character varying(50) NULL,
+    website character varying(500) NULL,
```
Comment (xenry) [inline db/migrations/campaign/V4.0.0.4__table_campaigns.sql:11]: I am wondering, maybe better store these additional properties like email, phone, website in a jsonb column? What do you think @kachar @igoychev ?

File: db/migrations/campaign/V4.0.0.4__table_campaigns.sql [hunk 3]
```diff
@@ -0,0 +1,21 @@
+BEGIN TRANSACTION;
+
+CREATE TABLE IF NOT EXISTS campaign.beneficiaries (
+    id uuid NOT NULL DEFAULT (gen_random_uuid()),
+    name character varying(100) NOT NULL,
```
Comment (kachar) [inline db/migrations/campaign/V4.0.0.4__table_campaigns.sql:5]: We can split the `name` in `first_name`, `last_name` to ensure that we can use them separatelly.

File: db/migrations/campaign/V4.0.0.4__table_campaigns.sql (hunk 2 above)
Comment (kachar) [reply db/migrations/campaign/V4.0.0.4__table_campaigns.sql:11]: I think `email` is quite important to be present as indexable column with strict type check.

It's going to be nice to have additional props as jsonb column as you say to collect metadata about the user that requires no special db structure as we can't properly type the elements of a json field.

An example might be the `website` field which may have a type check for `http://` or `https://` which will guarantee db integrity.

File: db/migrations/campaign/V4.0.0.4__table_campaigns.sql [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    type integer NOT NULL,
+    iso2country_code integer NOT NULL,
+    city character varying(50) NOT NULL,
```
Comment (kachar) [inline db/migrations/campaign/V4.0.0.4__table_campaigns.sql:8]: Using string `city` might lead to data duplication for free form entries like `Sofia`, `София`, etc

File: db/migrations/campaign/V4.0.0.4__table_campaigns.sql [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    email character varying(100) NULL,
+    phone character varying(50) NULL,
+    website character varying(500) NULL,
+	
+    CONSTRAINT "PK_Beneficiaries" PRIMARY KEY (id)
+);
+
+ALTER TABLE campaign.campaigns
+ADD CONSTRAINT "FK_Campaigns_Beneficiaries_BeneficiaryId" FOREIGN KEY (beneficiary_id) REFERENCES campaign.beneficiaries (id) ON DELETE CASCADE;
```
Comment (kachar) [inline db/migrations/campaign/V4.0.0.4__table_campaigns.sql:17]: We can add `unique` contstraint to the `email` column so we don't have duplicate beneficiary records for the same person

File: backend/protos/campaign.proto (hunk 1 above)
Comment (mayapeneva) [reply backend/protos/campaign.proto:91]: Done.

File: db/migrations/campaign/V4.0.0.4__table_campaigns.sql [hunk 6]
```diff
@@ -0,0 +1,21 @@
+BEGIN TRANSACTION;
+
+CREATE TABLE IF NOT EXISTS campaign.beneficiaries (
+    id uuid NOT NULL DEFAULT (gen_random_uuid()),
+    name character varying(100) NOT NULL,
+    type integer NOT NULL,
+    iso2country_code integer NOT NULL,
```
Comment (tanyogeorgiev) [inline db/migrations/campaign/V4.0.0.4__table_campaigns.sql:7]: We can store country code as is (string representation). This number is not very self-explanatory.

File: db/migrations/campaign/V4.0.0.4__table_campaigns.sql (hunk 3 above)
Comment (mayapeneva) [reply db/migrations/campaign/V4.0.0.4__table_campaigns.sql]: Done.

File: db/migrations/campaign/V4.0.0.4__table_campaigns.sql (hunk 2 above)
Comment (kachar) [reply db/migrations/campaign/V4.0.0.4__table_campaigns.sql:12]: Of course these ideas above are not mandatory for this iteration, we can implement them in the next sprints as well.

File: backend/module-campaigns/src/Podkrepibg.Campaigns.Application/Beneficiaries/Commands/CreateBeneficiary/CreateBeneficiaryValidator.cs [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    using FluentValidation;
+
+    public class CreateBeneficiaryValidator : AbstractValidator<CreateBeneficiaryCommand>
+    {
+        public CreateBeneficiaryValidator()
+        {
+            RuleFor(b => b.Request.FirstName)
+                .NotEmpty();
+
+            RuleFor(b => b.Request.LastName)
```
Comment (DeyanAtanasov) [inline backend/module-campaigns/src/Podkrepibg.Campaigns.Application/Beneficiaries/Commands/CreateBeneficiary/CreateBeneficiaryValidator.cs:13]: Should we not use MaximumLength to validate whether the last name is longer than the database column? If we decide to do it/use it we should probably do that for all of the string values.

**Pull request merged**